include/sddf/util/string.h

```diff
@@ -54,101 +44,76 @@ static inline char *sddf_strncpy(char *dest, const char *restrict src,
         *to++ = '\0';
     }
     return dest;
-#endif
 }
 
 static inline int sddf_strcmp(const char *a, const char *b)
 {
-#if __has_builtin(__builtin_strcmp)
-    return __builtin_strcmp(a, b);
-#else
     while (*a != '\0' && *b != '\0' && *a == *b) {
         a++;
         b++;
     }
     return (int)(*a) - *b;
-#endif
 }
 
 static inline int sddf_strncmp(const char *a, const char *b, size_t n)
 {
-#if __has_builtin(__builtin_strncmp)
-    return __builtin_strncmp(a, b, n);
-#else
     for (size_t i = 0; i < n; i++) {
         if (a[i] == '\0' || b[i] == '\0' || a[i] != b[i]) {
             return (int)a[i] - b[i];
         }
     }
     return 0;
-#endif
 }
 
 static inline char *sddf_strchr(const char *s, int c)
 {
-#if __has_builtin(__builtin_strncmp)
-    return __builtin_strchr(s, c);
-#else
     while (*s != '\0') {
         if (*s == c) {
-            return s;
+            return (char *)s;
         }
         s++;
     }
     if (c == '\0') {
-        return s;
+        return (char *)s;
     }
     return NULL;
-#endif
 }
 
 static inline int sddf_memcmp(const void *a, const void *b, size_t n)
 {
-#if __has_builtin(__builtin_memcmp)
-    return __builtin_memcmp(a, b, n);
-#else
-    unsigned char *_a = a;
-    unsigned char *_b = b;
+    unsigned char *_a = (unsigned char *)a;
```

Comment:
These casts should not be necessary.  `void *` is assignment compatible with `unsigned char *`